packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -92,15 +169,126 @@ class FlutterGoldenFileComparator implements GoldenFileComparator {
     }
     return true;
   }
+}
+
+/// A [FlutterGoldenFileComparator] for testing golden images with Skia Gold.
+///
+/// For testing across all platforms, the [SkiaGoldClient] is used to upload
+/// images for framework-related golden tests and process results. Currently
+/// these tests are designed to be run post-submit on Cirrus CI, informed by the
+/// environment.
+///
+/// See also:
+///
+///  * [GoldenFileComparator], the abstract class that
+///    [FlutterGoldenFileComparator] implements.
+///  * [FlutterGoldensRepositoryFileComparator], another
+///    [FlutterGoldenFileComparator] that tests golden images using the
+///    flutter/goldens repository.
+class FlutterSkiaGoldFileComparator extends FlutterGoldenFileComparator {
+  FlutterSkiaGoldFileComparator(
+    Uri basedir, {
+    FileSystem fs = const LocalFileSystem(),
+    Platform platform = const LocalPlatform(),
+  }) : super(
+    basedir,
+    fs: fs,
+    platform: platform,
+  ) {
+    _skiaClient = SkiaGoldClient(fs.directory(basedir));
+  }
+
+  SkiaGoldClient _skiaClient;
+
+  /// Creates a new [FlutterSkiaGoldFileComparator] that mirrors the relative
+  /// path resolution of the default [goldenFileComparator].
+  ///
+  /// The [goldens] and [defaultComparator] parameters are visible for testing
+  /// purposes only.
+  static Future<FlutterSkiaGoldFileComparator> fromDefaultComparator({
+    GoldensClient goldens,
+    LocalFileComparator defaultComparator,
+  }) async {
+    defaultComparator ??= goldenFileComparator;
+    goldens ??= GoldensClient();
+
+    final Directory baseDirectory = FlutterGoldenFileComparator.getBaseDirectory(goldens, defaultComparator);
+    if (!baseDirectory.existsSync())
+      baseDirectory.createSync(recursive: true);
+    final SkiaGoldClient skiaClient = SkiaGoldClient(baseDirectory);
+    await skiaClient.auth();
+    await skiaClient.imgtestInit();
+    return FlutterSkiaGoldFileComparator(baseDirectory.uri);
+  }
 
   @override
-  Future<void> update(Uri golden, Uint8List imageBytes) async {
-    final File goldenFile = _getGoldenFile(golden);
-    await goldenFile.parent.create(recursive: true);
-    await goldenFile.writeAsBytes(imageBytes, flush: true);
+  Future<bool> compare(Uint8List imageBytes, Uri golden) async {
+    golden = _addPrefix(golden);
+    await update(golden, imageBytes);
```

Comment:
How come we `update()` during `compare()`?

Is it that we update the version in Gold that's associated with this test run whenever we compare, and that's separate from approving which is the new golden?